version: '3.5'
services:
  api1: &api 
    build: .
    container_name: rinha-api1
    hostname: rinha-api1
    restart: always
    environment:
      DB_URL: "postgres://postgres:postgres@rinha-db:5432/rinhadb"
    depends_on:
      - db
    expose:
      - "3000"
    deploy:
      resources:
        limits:
          cpus: '0.4'
          memory: '0.8GB'

  api2: 
    <<: *api
    container_name: rinha-api2
    hostname: rinha-api2

  nginx:
    container_name: rinha-nginx
    image: nginx:alpine
    restart: always
    depends_on:
      - api1
      - api2
    ports:
        - "9999:9999"
    volumes:
        - ./nginx.conf:/etc/nginx/nginx.conf
    deploy:
      resources:
          limits:
              cpus: '0.10'
              memory: '0.1GB'

  db: # Banco de dados
    image: postgres
    container_name: rinha-db
    hostname: rinha-db
    restart: always
    command: postgres -c 'max_connections=450' -c 'log_error_verbosity=terse'
    environment:
      POSTGRES_PASSWORD: postgres
      POSTGRES_USER: postgres
      POSTGRES_DB: rinhadb
    ports:
      - "5432:5432"
    expose:
      - "5432"
    deploy:
      resources:
        limits:
          cpus: '0.6'
          memory: '1.3GB'
    volumes:
      - pg_data:/var/lib/postgresql/data
      - ./db.sql:/docker-entrypoint-initdb.d/db.sql
volumes:
  pg_data:
    driver: local
networks:
  default:
    driver: bridge
    name: rinha